testApi/src/test/scala/africa/absa/testing/scapi/json/SuiteFactoryTest.scala

```diff
@@ -129,43 +130,43 @@ class SuiteFactoryTest extends FunSuite {
    */
   test("filterOnlyOrAll - only used - once") {
     val suitesBundles = Set(
-      SuiteBundle(suite = Suite(endpoint = "endpoint1", tests = Set(
-        SuiteTestScenario(name = "test1", categories = Set("SMOKE"), headers = Set.empty, actions = Set.empty, responseActions = Set.empty, only = Some(false)),
-        SuiteTestScenario(name = "test2", categories = Set("SMOKE"), headers = Set.empty, actions = Set.empty, responseActions = Set.empty, only = Some(true))
+      Suite(suite = TestSet(name = "name1", tests = Set(
+        SuiteTestScenario(name = "test1", categories = Seq("SMOKE"), headers = Seq.empty, actions = Seq.empty, responseActions = Seq.empty, only = Some(false)),
+        SuiteTestScenario(name = "test2", categories = Seq("SMOKE"), headers = Seq.empty, actions = Seq.empty, responseActions = Seq.empty, only = Some(true))
       ))),
-      SuiteBundle(suite = Suite(endpoint = "endpoint1", tests = Set(
-        SuiteTestScenario(name = "test1", categories = Set("SMOKE"), headers = Set.empty, actions = Set.empty, responseActions = Set.empty, only = Some(false)),
-        SuiteTestScenario(name = "test2", categories = Set("SMOKE"), headers = Set.empty, actions = Set.empty, responseActions = Set.empty, only = Some(true))
+      Suite(suite = TestSet(name = "name1", tests = Set(
+        SuiteTestScenario(name = "test1", categories = Seq("SMOKE"), headers = Seq.empty, actions = Seq.empty, responseActions = Seq.empty, only = Some(false)),
+        SuiteTestScenario(name = "test2", categories = Seq("SMOKE"), headers = Seq.empty, actions = Seq.empty, responseActions = Seq.empty, only = Some(true))
       ))),
-      SuiteBundle(suite = Suite(endpoint = "endpoint2", tests = Set(
-        SuiteTestScenario(name = "test1", categories = Set("SMOKE"), headers = Set.empty, actions = Set.empty, responseActions = Set.empty, only = Some(false)),
+      Suite(suite = TestSet(name = "name2", tests = Set(
+        SuiteTestScenario(name = "test1", categories = Seq("SMOKE"), headers = Seq.empty, actions = Seq.empty, responseActions = Seq.empty, only = Some(false)),
       ))))
 
-    val filteredSuiteBundles: Set[SuiteBundle] = SuiteFactory.filterOnlyOrAll(suitesBundles)
+    val filteredSuiteBundles: Set[Suite] = SuiteFactory.filterOnlyOrAll(suitesBundles)
 
     assertEquals(filteredSuiteBundles.size, 1)
 
     val filteredSuite = filteredSuiteBundles.head.suite
-    assertEquals(filteredSuite.endpoint, "endpoint1")
-    assertEquals(filteredSuite.tests.size, 1)
+    assertEquals(clue("name1"), clue(filteredSuite.name))
```

Comment:
What is the point of calling `clue` here (and everywhere else), sorry, I don't get it

Reply:
I found an error made from several test changes. Correct usage of Clue is in simple boolean assert method call.

- All location with clue used were refactored.
- Why to use it? Here is comparison of output "when error happen":

**without clue**
<img width="550" alt="image" src="https://github.com/AbsaOSS/ScAPI/assets/109138042/1924d57c-e0f3-4aa5-8726-625f57e3f133">

**with clue** 
<img width="540" alt="image" src="https://github.com/AbsaOSS/ScAPI/assets/109138042/56f31993-7cdb-44c4-adb0-6cac2bcdfd73">

You can see that clue method provide additional information. It can be useful for  complex objects.

Reply:
Thanks, got it.